### PR TITLE
Refactor ca bundle injection for maintainability

### PR DESCRIPTION
**Pull request by marun**
Previously injecting ca bundles into configmap and apiservices required running separate processes. With the need to [add injection support for mutating and validating webhookconfigurations](https://github.com/openshift/service-ca-operator/pull/51), it was time to factor out common code and allow a single process to run controllers for all supported injection targets.

### CODE REVIEW

**Comment by marun** [comment]
/retest

e2e-aws failure should be fixed by the merge of https://github.com/openshift/origin/pull/24085

File: pkg/controller/cabundleinjector/starter.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	scsv1alpha1 "github.com/openshift/api/servicecertsigner/v1alpha1"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/service-ca-operator/pkg/controller/api"
+)
+
+type commonInjectorConfig struct {
```
Comment (stlaz) [inline pkg/controller/cabundleinjector/starter.go]: nit: `caBundleInjectorConfig`?

**Comment by marun** [comment]
I'm going to wait to squash until openshift/api#506 merges and I can update to point to master instead of the PR branch.

**Comment by marun** [comment]
Rebased

**Comment by marun** [comment]
Rebased on #88, which will need to merge before this PR.

**Comment by marun** [comment]
Rebased and ready for review

**Comment by marun** [comment]
Rebased to pick up the extended polling timeout

**Comment by sttts** [comment]
Unrelated flakes.

/retest
/lgtm
/approve

**Pull request merged**